### PR TITLE
Change structure acheievements

### PR DESCRIPTION
**Pull request by ospodaryk**
# GreenCity PR
[Update Achievement structure #6421](https://github.com/ita-social-projects/GreenCity/issues/6421)

## Summary Of Changes :fire:
Added new enum of achievements
Updated old implementation of achievement service

## Added
New enum of achievements

## Changed
Implementation of achievement service

## Deleted
Unused methods and fields

# PR Checklist Forms

_(to be filled out by PR submitter)_
- [ ] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: core/src/main/java/greencity/controller/AchievementController.java [hunk 1]
```diff
@@ -47,39 +43,4 @@ public AchievementController(AchievementService achievementService) {
...
-     * @param userId of {@link UserVO}
-     * @return list {@link AchievementNotification}
-     */
-    @ApiOperation(value = "Get all the achievements that need to notify.")
-    @ApiResponses(value = {
-        @ApiResponse(code = 200, message = HttpStatuses.OK),
-        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
-        @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
-    })
-    @GetMapping("/notification/{userId}")
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/AchievementController.java:63]: are you sure that UI components do not use these endpoints?

File: service/src/test/java/greencity/ModelUtils.java [hunk 2]
```diff
@@ -444,7 +443,7 @@ public static UserVO getUserVOWithData() {
             .userAchievements(List.of(
                 UserAchievementVO.builder()
                     .id(47L)
-                    .achievementStatus(AchievementStatus.ACTIVE)
+
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/ModelUtils.java]: please remove this blank line

File: service-api/src/main/java/greencity/constant/ErrorMessage.java [hunk 3]
```diff
@@ -157,7 +159,9 @@ public final class ErrorMessage {
         "Count of tags should be at least one but not more three";
     public static final String TOKEN_FOR_RESTORE_IS_INVALID = "Token is null or it doesn't exist.";
     public static final String ACHIEVEMENT_NOT_DELETED = "Achievement not deleted ";
-    public static final String ACHIEVEMENT_NOT_FOUND_BY_ID = "The name does not exist by this id: ";
+    public static final String ACHIEVEMENT_NOT_FOUND_BY_ID = "The achievement does not exist by this id: ";
+    public static final String ACHIEVEMENT_CATEGORY_NOT_FOUND_BY_ID =
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/constant/ErrorMessage.java]: Where do you use it?

File: service-api/src/main/java/greencity/constant/ErrorMessage.java [hunk 4]
```diff
@@ -1,6 +1,8 @@
 package greencity.constant;
 
 public final class ErrorMessage {
+    public static final String RATING_CALCULATION_ENUM_NOT_FOUND_BY_NAME =
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/constant/ErrorMessage.java]: Where do you use it?

File: core/src/main/java/greencity/controller/AchievementController.java (hunk 1 above)
Comment (ospodaryk) [reply core/src/main/java/greencity/controller/AchievementController.java:63]: Yes

File: service-api/src/main/java/greencity/constant/ErrorMessage.java (hunk 4 above)
Comment (ospodaryk) [reply service-api/src/main/java/greencity/constant/ErrorMessage.java]: deleted

File: service-api/src/main/java/greencity/constant/ErrorMessage.java (hunk 3 above)
Comment (ospodaryk) [reply service-api/src/main/java/greencity/constant/ErrorMessage.java]: deleted

File: service/src/test/java/greencity/ModelUtils.java (hunk 2 above)
Comment (ospodaryk) [reply service/src/test/java/greencity/ModelUtils.java]: deleted

File: service/src/test/java/greencity/ModelUtils.java [hunk 5]
```diff
@@ -9,7 +9,7 @@
 import greencity.dto.PageableAdvancedDto;
 import greencity.dto.achievement.AchievementManagementDto;
 import greencity.dto.achievement.AchievementPostDto;
-import greencity.dto.achievement.AchievementTranslationVO;
+
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/ModelUtils.java]: please remove this blank line

File: service/src/test/java/greencity/ModelUtils.java [hunk 6]
```diff
@@ -162,11 +162,10 @@
 import greencity.entity.event.EventComment;
 import greencity.entity.event.EventDateLocation;
 import greencity.entity.event.EventGrade;
-import greencity.entity.localization.AchievementTranslation;
+
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/ModelUtils.java]: same here

File: service/src/test/java/greencity/ModelUtils.java [hunk 7]
```diff
@@ -454,7 +452,7 @@ public static UserVO getUserVOWithData() {
                     .build(),
                 UserAchievementVO.builder()
                     .id(39L)
-                    .achievementStatus(AchievementStatus.INACTIVE)
+
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/ModelUtils.java]: same here

File: service/src/test/java/greencity/ModelUtils.java [hunk 8]
```diff
@@ -1379,48 +1377,44 @@ public static AdvicePostDto getAdvicePostDto() {
     }
 
     public static Achievement getAchievement() {
-        return new Achievement(1L, Collections.singletonList(getAchievementTranslation()), Collections.emptyList(),
+        return new Achievement(1L,
+            "CREATED_5_NEWS", "CREATED_5_NEWS", "CREATED_5_NEWS",
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/ModelUtils.java]: it is not a good practice to use the same value for the different variables

File: service/src/test/java/greencity/ModelUtils.java [hunk 9]
```diff
@@ -1379,48 +1377,44 @@ public static AdvicePostDto getAdvicePostDto() {
...
 
     public static AchievementVO getAchievementVO() {
-        return new AchievementVO(1L, Collections.emptyList(), Collections.emptyList(), new AchievementCategoryVO(), 1);
+        return new AchievementVO(1L, "CREATED_5_NEWS", "CREATED_5_NEWS", "CREATED_5_NEWS", new AchievementCategoryVO(),
+            1);
     }
 
     public static AchievementPostDto getAchievementPostDto() {
-        return new AchievementPostDto(Collections.emptyList(), getAchievementCategoryDto(), 1);
+        return new AchievementPostDto("CREATED_5_NEWS", "CREATED_5_NEWS", "CREATED_5_NEWS", getAchievementCategoryDto(),
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/ModelUtils.java]: same here

File: service/src/test/java/greencity/ModelUtils.java [hunk 10]
```diff
@@ -1379,48 +1377,44 @@ public static AdvicePostDto getAdvicePostDto() {
...
     }
 
     public static AchievementCategory getAchievementCategory() {
-        return new AchievementCategory(1L, "Name", null, null);
+        return new AchievementCategory(1L, "Name", Collections.emptyList());
     }
 
     public static AchievementVO getAchievementVO() {
-        return new AchievementVO(1L, Collections.emptyList(), Collections.emptyList(), new AchievementCategoryVO(), 1);
+        return new AchievementVO(1L, "CREATED_5_NEWS", "CREATED_5_NEWS", "CREATED_5_NEWS", new AchievementCategoryVO(),
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/ModelUtils.java]: same here

File: service/src/test/java/greencity/ModelUtils.java [hunk 11]
```diff
@@ -1698,7 +1692,7 @@ public static UserVO createUserVO2() {
                     .userAchievements(List.of(
                         UserAchievementVO.builder()
                             .id(47L)
-                            .achievementStatus(AchievementStatus.ACTIVE)
+
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/ModelUtils.java]: please remove blank line

File: service/src/test/java/greencity/ModelUtils.java [hunk 12]
```diff
@@ -1708,7 +1702,7 @@ public static UserVO createUserVO2() {
                             .build(),
                         UserAchievementVO.builder()
                             .id(39L)
-                            .achievementStatus(AchievementStatus.INACTIVE)
+
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/ModelUtils.java]: same here

File: service/src/test/java/greencity/ModelUtils.java [hunk 13]
```diff
@@ -1776,7 +1770,7 @@ public static UserVO createUserVO2() {
             .userAchievements(List.of(
                 UserAchievementVO.builder()
                     .id(47L)
-                    .achievementStatus(AchievementStatus.ACTIVE)
+
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/ModelUtils.java]: same here

File: service/src/test/java/greencity/ModelUtils.java [hunk 14]
```diff
@@ -1786,7 +1780,7 @@ public static UserVO createUserVO2() {
                     .build(),
                 UserAchievementVO.builder()
                     .id(39L)
-                    .achievementStatus(AchievementStatus.INACTIVE)
+
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/ModelUtils.java]: same here

File: service/src/test/java/greencity/service/AchievementServiceImplTest.java [hunk 15]
```diff
@@ -7,33 +7,28 @@
...
 import greencity.dto.achievementcategory.AchievementCategoryVO;
-import greencity.dto.language.LanguageVO;
 import greencity.dto.user.UserVO;
 import greencity.dto.useraction.UserActionVO;
 import greencity.entity.Achievement;
 import greencity.entity.AchievementCategory;
-import greencity.entity.Language;
-import greencity.entity.UserAchievement;
-import greencity.entity.localization.AchievementTranslation;
+
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/AchievementServiceImplTest.java]: please remove blank line

File: service/src/test/java/greencity/service/AchievementServiceImplTest.java [hunk 16]
```diff
@@ -7,33 +7,28 @@
...
 import greencity.enums.AchievementCategoryType;
+
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/AchievementServiceImplTest.java]: same here

File: service/src/test/java/greencity/service/EcoNewsServiceImplTest.java [hunk 17]
```diff
@@ -122,14 +127,15 @@ void saveWithExistedImage() throws IOException {
...
-
         when(modelMapper.map(addEcoNewsDtoRequest, EcoNews.class)).thenReturn(ecoNews);
         when(restClient.findByEmail(TestConst.EMAIL)).thenReturn(ModelUtils.getUserVO());
         when(fileService.upload(any(MultipartFile.class))).thenReturn(ModelUtils.getUrl().toString());
         List<TagVO> tagVOList = Collections.singletonList(ModelUtils.getTagVO());
         when(tagService.findTagsByNamesAndType(anyList(), eq(TagType.ECO_NEWS))).thenReturn(tagVOList);
         when(ecoNewsRepo.save(any(EcoNews.class))).thenReturn(ecoNews);
+        addEcoNewsDtoResponse.setEcoNewsAuthorDto(ModelUtils.getEcoNewsAuthorDto());
         when(modelMapper.map(ecoNews, AddEcoNewsDtoResponse.class)).thenReturn(addEcoNewsDtoResponse);
+        when(modelMapper.map(ModelUtils.getUserVO(), User.class)).thenReturn(ModelUtils.getUser());
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/EcoNewsServiceImplTest.java:139]: please add `verify`

File: service/src/test/java/greencity/service/EcoNewsServiceImplTest.java [hunk 18]
```diff
@@ -178,7 +184,7 @@ void saveEcoNews() {
         when(modelMapper.map(tagVOList,
             new TypeToken<List<Tag>>() {
             }.getType())).thenReturn(tags);
-
+        when(userService.findByEmail(anyString())).thenReturn(ModelUtils.getUserVO());
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/EcoNewsServiceImplTest.java:188]: same here

File: service/src/test/java/greencity/service/HabitAssignServiceImplTest.java [hunk 19]
```diff
@@ -469,6 +473,7 @@ void unenrollHabit() {
             .thenReturn(Optional.of(habitAssign));
         when(habitStatusCalendarRepo.findHabitStatusCalendarByEnrollDateAndHabitAssign(date, habitAssign))
             .thenReturn(habitStatusCalendar);
+        when(userService.findById(any())).thenReturn(ModelUtils.getUserVO());
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/HabitAssignServiceImplTest.java:476]: please add `verify`

File: service/src/test/java/greencity/service/HabitAssignServiceImplTest.java [hunk 20]
```diff
@@ -558,6 +563,7 @@ void deleteHabitAssign() {
         habitAssign.setWorkingDays(10);
 
         when(habitAssignRepo.findById(habitAssignId)).thenReturn(Optional.of(habitAssign));
+        when(userService.findById(any())).thenReturn(ModelUtils.getUserVO());
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/HabitAssignServiceImplTest.java:567]: same here

File: service/src/test/java/greencity/service/HabitAssignServiceImplTest.java [hunk 21]
```diff
@@ -884,6 +890,7 @@ void enrollHabit() {
                 .thenReturn(null);
         when(modelMapper.map(habitAssign, HabitAssignDto.class)).thenReturn(habitAssignDto);
         when(modelMapper.map(translation, HabitDto.class)).thenReturn(habitDto);
+        when(userService.findById(any())).thenReturn(ModelUtils.getUserVO());
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/HabitAssignServiceImplTest.java:895]: same here

**Review by RomaMocherniuk — APPROVED** [review]

**Review by HelenSotnik — APPROVED** [review]

**Review by ABbondar — APPROVED** [review]

**Review by Maksym-Lenets — APPROVED** [review]

**Pull request merged**